content/2018-10-31-xsimd.rst

```diff
@@ -0,0 +1,271 @@
+
+.. code::
+
+    All benchmarks:
+
+        before     after       ratio
+      [99d8234f] [60632651]
+         9.90μs     9.89μs      1.00  benchmarks.TimeSuite.time_abs_array
+    +   36.82μs    58.44μs      1.59  benchmarks.TimeSuite.time_acos_array
+        36.25μs    33.60μs      0.93  benchmarks.TimeSuite.time_asin_array
+    -   50.47μs    33.03μs      0.65  benchmarks.TimeSuite.time_atan_array
+    -   48.62μs    35.72μs      0.73  benchmarks.TimeSuite.time_cos_array
+    -   73.82μs    43.81μs      0.59  benchmarks.TimeSuite.time_cosh_array
+    -   47.55μs    35.52μs      0.75  benchmarks.TimeSuite.time_sin_array
+    -   91.45μs    47.86μs      0.52  benchmarks.TimeSuite.time_sinh_array
+        18.35μs    17.91μs      0.98  benchmarks.TimeSuite.time_sqrt_array
+         9.60μs    10.05μs      1.05  benchmarks.TimeSuite.time_square_array
+    -   71.71μs    33.35μs      0.47  benchmarks.TimeSuite.time_tan_array
+    -   84.63μs    42.28μs      0.50  benchmarks.TimeSuite.time_tanh_array
+
+Looks pretty good! Apart from a regression on ``acos``, this is either on-par or faster than before.
+
+Out of curiosity, I also ran the same benchmark, but using Clang as back-end.
+
+.. code::
+
+    All benchmarks:
+
+        before     after       ratio
+      [99d8234f] [60632651]
+         9.57μs    10.00μs      1.05  benchmarks.TimeSuite.time_abs_array
+    +   34.20μs    58.53μs      1.71  benchmarks.TimeSuite.time_acos_array
+        36.09μs    33.91μs      0.94  benchmarks.TimeSuite.time_asin_array
+    -   45.02μs    33.86μs      0.75  benchmarks.TimeSuite.time_atan_array
+    +   39.44μs    45.48μs      1.15  benchmarks.TimeSuite.time_cos_array
+    -   65.98μs    44.78μs      0.68  benchmarks.TimeSuite.time_cosh_array
+    +   39.39μs    45.48μs      1.15  benchmarks.TimeSuite.time_sin_array
+    -  110.62μs    48.44μs      0.44  benchmarks.TimeSuite.time_sinh_array
+        18.18μs    18.54μs      1.02  benchmarks.TimeSuite.time_sqrt_array
+        10.05μs     9.56μs      0.95  benchmarks.TimeSuite.time_square_array
+    -   56.82μs    45.32μs      0.80  benchmarks.TimeSuite.time_tan_array
+    -   98.85μs    44.16μs      0.45  benchmarks.TimeSuite.time_tanh_array
+
+Wow, that's significant changes. Regression on both ``cos``, ``sin`` and ``acos`` are not good news.
+
+What conclusion should we draw? My take on this is that these benchmarks are
+not synthetic enough to state *xsimd implementation of function X is better or
+worse than boost.simd implementation*. But maybe there are bad interactions
+with Pythran's expression template? A single register spill could wreak havoc
+in the performance, and I know there is room for improvement there.
+
+Conclusions
+===========
+
+I'm indeed very happy with the changes. The xsimd team is very reactive, it's
+cool to chat with them about performance, Python, C++... And did I see xsimd
+supports Neon, avx512? I should try to run cross-compiled Pythran code on a
```

Comment:
Upstream capitalization is NEON and AVX512 (not sure why for NEON actually, since that is never spelled out as an acronym anywhere in ARM's documentation, but upstream is always right :smiley: )